src/lib.rs

```diff
@@ -51,9 +57,300 @@ mod writer;
 
 pub use parser::ParserError;
 
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub enum ParseEnumError {
+    Invalid(String),
+}
+
+/// Type of network of the originator or a connection of the session.
+///
+/// See [RFC 8866 Section 5.2](https://tools.ietf.org/html/rfc8866#section-5.2),
+/// [RFC 8866 Section 5.7](https://tools.ietf.org/html/rfc8866#section-5.7) and
+/// [RFC 8866 Section 8.2.6](https://datatracker.ietf.org/doc/html/rfc8866#section-8.2.6) for more details
+#[derive(Debug, PartialEq, Eq, Clone, Copy)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+pub enum NetType {
+    //Internet
+    In,
+    // Telephone Network
+    Tn,
+    // ATM Bearer Connection
+    Atm,
+    // Public Switched Telephone Network
+    Pstn,
+}
+
+impl NetType {
+    pub fn as_str(&self) -> &'static str {
+        match self {
+            NetType::In => "IN",
+            NetType::Tn => "TN",
+            NetType::Atm => "ATM",
+            NetType::Pstn => "PSTN",
+        }
+    }
+}
+
+impl FromStr for NetType {
+    type Err = ParseEnumError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
```

Comment:
Should probably use

```suggestion
        match s.to_ascii_lowercase() {
```

here or `eq_ignore_ascii_case()` as everything except for a handful of strings are case-insensitive according to the RFC

Reply:
All strings that are case-sensitive are explicitly mentioned like that in the RFC (in the grammar in section 9)